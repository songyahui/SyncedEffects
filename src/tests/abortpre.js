"use hopscript"

var hh = require("hiphop");

var prg =
    <hh.module O S>
      <hh.loop>
	<hh.abort pre S>
	  <hh.emit S/>
	  <hh.pause/>
	  <hh.emit O/>
	</hh.abort>
	<hh.pause/>
      </hh.loop>
    </hh.module>;

//console.error(prg.pretty_print())

exports.prg = new hh.ReactiveMachine(prg, "abortpre");
