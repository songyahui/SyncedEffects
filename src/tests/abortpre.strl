module abortpre:

output O, S;

loop
	abort
		emit S;
		pause;
		emit O;
	when pre(S);
	pause;
end loop

end module