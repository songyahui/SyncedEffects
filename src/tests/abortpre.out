abortpre> ;
--- Output: S
abortpre> ;
--- Output:
abortpre> ;
--- Output: S
abortpre> ;
--- Output:
abortpre> ;
--- Output: S
